### PR TITLE
[3889][next] Remove all usage of `environment-config.xml`

### PR DESCRIPTION
**Pull request by jvega190**
3.1.7 Ticket
https://github.com/craftercms/craftercms/issues/3889

### CODE REVIEW

**Review by rart — CHANGES_REQUESTED** [review]
Don't remove the variables. These are used all over the place. Replace the values.
The task is not to remove `envConfig` uses but replace the references to `environment-config.xml`

**Comment by jvega190** [comment]
@rart variables were not removed, they were just being set some lines after, but it was not necessary so I moved them back to where they were (using location.origin)

**Comment by sumerjabri** [comment]
@jvega190 this needs to go to 3.1.x as draft, and as non-draft for `develop`

File: templates/web/common/js-global-context.ftl [hunk 1]
```diff
@@ -73,8 +73,6 @@
   },
   </#if>
   "env": {
-    "AUTHORING_BASE": "${env_config.authoringServerUrl!"/studio"}",
-    "GUEST_BASE": "${env_config.previewServerUrl!"http://localhost:8080"}",
```
Comment (rart) [inline templates/web/common/js-global-context.ftl:77]: Rollback

File: ui/app/src/state/store.ts [hunk 2]
```diff
@@ -37,9 +37,20 @@ epicMiddleware.run(epic);
...
+
+      // Setting AUTHORING AND GUEST BASE, node dev will have its values already set pointing to 8080
+      if (!state.env.AUTHORING_BASE) {
+        state.env.AUTHORING_BASE = window.location.origin + '/studio';
+      }
+
+      if (!state.env.GUEST_BASE) {
+        state.env.GUEST_BASE = window.location.origin;
+      }
+
```
Comment (rart) [inline ui/app/src/state/store.ts]: Rollback

File: templates/web/common/js-global-context.ftl (hunk 1 above)
Comment (rart) [inline templates/web/common/js-global-context.ftl:77]: Rollback

**Review by rart — CHANGES_REQUESTED** [review]
- Freemarker files got misformatted
- For the overlayhook & js-global-context changes, let's discuss. Would want those things to come from the freemarker and/or do something more centralized.

File: templates/web/common/js-global-context.ftl [hunk 3]
```diff
@@ -16,6 +16,8 @@
 
 <#assign loggedIn = ((userEmail?? && userEmail != "") || (envConfig??))/>
 <#assign env_config = envConfig???then(envConfig + _csrf, {})/>
+<#assign locationOrigin = '${request.scheme}://${request.serverName}:${request.serverPort?c}' />
```
Comment (rart) [inline templates/web/common/js-global-context.ftl:19]: What if there’s no port? (e.g. :80)

File: templates/web/common/page-fragments/studio-context.ftl [hunk 4]
```diff
@@ -27,33 +27,33 @@
...
-		baseUri: "/studio",
-		authoringAppBaseUri: "${envConfig.authoringServerUrl!'/studio'}",
-		formServerUri: "${envConfig.formServerUrl!'UNSET'}",
-		previewAppBaseUri: "${envConfig.previewServerUrl!'UNSET'}",
-		contextMenuOffsetPage: false,
-		brandedLogoUri: "/api/1/services/api/1/content/get-content-at-path.bin?path=/configuration/app-logo.png",
-		homeUri: "/site-dashboard?site=${envConfig.site!'UNSET'}",
+    collabSandbox: '',
+    baseUri: '/studio',
+    authoringAppBaseUri: window.location.origin + '/studio',
```
Comment (rart) [inline templates/web/common/page-fragments/studio-context.ftl:32]: Same work here

File: templates/web/overlayhook.ftl [hunk 5]
```diff
@@ -1,6 +1,8 @@
-<#assign authoringServer = envConfig.authoringServer!''/>
+<#assign locationOrigin = '${request.scheme}://${request.serverName}:${request.serverPort?c}' />
```
Comment (rart) [inline templates/web/overlayhook.ftl:1]: What if there’s no port? (e.g. :80)

**Review by rart — CHANGES_REQUESTED** [review]

**Comment by rart** [comment]
Closed in favour of https://github.com/craftercms/studio-ui/pull/1724

**Pull request closed without merging**